src/methods.jl

```diff
@@ -1,5 +1,128 @@
 export FDMReport, fdm, backward_fdm, forward_fdm, central_fdm
 
+"""
+    FDM.DEFAULT_CONDITION
+
+The default [condition number](https://en.wikipedia.org/wiki/Condition_number) used when
+computing bounds. It provides amplification of the ∞-norm when passed to the function's
+derivatives.
+"""
+const DEFAULT_CONDITION = 100
+
+"""
+    FDM.TINY
+
+A tiny number added to some quantities to ensure that division by 0 does not occur.
+"""
+const TINY = 1e-20
+
+forward_grid(p::Int) = 0:(p - 1)
+backward_grid(p::Int) = (1 - p):0
+function central_grid(p::Int)
+    if isodd(p)
+        return div(1 - p, 2):div(p - 1, 2)
+    else
+        return vcat(div(-p, 2):-1, 1:div(p, 2))
+    end
+end
+
+mutable struct History
+    adapt::Int
+    eps::Real
+    bound::Real
+    step::Real
+    accuracy::Real
```

Comment:
Do we care that the types of `eps`, `bound`, etc are abstract? Does this likely have performance implications?

Reply:
It does, but this is a mutable type anyway, so it will already not be terribly performant. The problem with having them concrete is that when the `History` object is constructed, we don't know the types of the values it will end up holding, since this is constructed before we call the `FDMethod` on a function. The function could return `Float32`, for example, and we wouldn't necessarily want to promote to `Float64`. (Though a case could be made to do so.)